javaparser-core/src/main/java/com/github/javaparser/JavaParser.java

```diff
@@ -112,8 +111,7 @@ private GeneratedJavaParser getParserForProvider(Provider provider) {
      * The start indicates what can be found in the source code (compilation unit, block, import...)
      *
      * @param start refer to the constants in ParseStart to see what can be parsed.
-     * @param provider refer to Providers to see how you can read source.
-     * The provider will be closed after parsing.
+     * @param provider refer to Providers to see how you can read source. The provider will be closed after parsing.
```

Comment:
Yes actually, there is always a stream inside somehow :) I get your point though.